crates/payload/builder/src/service.rs

```diff
@@ -303,7 +303,7 @@ where
         let (fut, keep_alive) = self.payload_jobs[job].0.resolve();
 
         if keep_alive == KeepPayloadJobAlive::No {
-            let (_, id) = self.payload_jobs.remove(job);
+            let (_, id) = self.payload_jobs.swap_remove(job);
```

Comment:
this is also fine because we always find the index first and the ordering doesn't matter